# HG changeset patch
# Parent 96ab26c1813537c2fe91c2aee6897b7f7cfab312
localmod: lib/Target/X86/X86ISelLowering.cpp lib/Target/X86/X86TargetObjectFile.cpp lib/Target/X86/X86TargetObjectFile.h

diff -r 96ab26c18135 -r dfa48e8496b6 lib/Target/X86/X86ISelLowering.cpp
--- a/lib/Target/X86/X86ISelLowering.cpp	Tue Aug 09 13:58:00 2011 -0700
+++ b/lib/Target/X86/X86ISelLowering.cpp	Tue Aug 09 14:31:29 2011 -0700
@@ -157,6 +157,10 @@
       return new X8664_MachoTargetObjectFile();
     return new TargetLoweringObjectFileMachO();
   }
+  // @LOCALMOD-START
+  if (Subtarget->isTargetNaCl()) {
+    return new X86NaClElfTargetObjectFile();
+  }
 
   if (Subtarget->isTargetELF())
     return new TargetLoweringObjectFileELF();
diff -r 96ab26c18135 -r dfa48e8496b6 lib/Target/X86/X86TargetObjectFile.cpp
--- a/lib/Target/X86/X86TargetObjectFile.cpp	Tue Aug 09 13:58:00 2011 -0700
+++ b/lib/Target/X86/X86TargetObjectFile.cpp	Tue Aug 09 14:31:29 2011 -0700
@@ -56,8 +56,8 @@
 // Without this the linker defined symbols __fini_array_start and
 // __fini_array_end do not have useful values. c.f.:
 // http://code.google.com/p/nativeclient/issues/detail?id=805
-void X8664_ELFTargetObjectFile::Initialize(MCContext &Ctx,
-                                           const TargetMachine &TM) {
+void X86_NaClELFTargetObjectFile::Initialize(MCContext &Ctx,
+                                             const TargetMachine &TM) {
   TargetLoweringObjectFileELF::Initialize(Ctx, TM);
   if (TM.getSubtarget<X86Subtarget>().isTargetNaCl()) {
     StaticCtorSection =
@@ -72,7 +72,7 @@
                                  SectionKind::getDataRel());
   }
 }
-
+#if 0
 void X8632_ELFTargetObjectFile::Initialize(MCContext &Ctx,
                                            const TargetMachine &TM) {
   TargetLoweringObjectFileELF::Initialize(Ctx, TM);
@@ -89,4 +89,5 @@
                                  SectionKind::getDataRel());
   }
 }
+#endif
 // @LOCALMOD-END
diff -r 96ab26c18135 -r dfa48e8496b6 lib/Target/X86/X86TargetObjectFile.h
--- a/lib/Target/X86/X86TargetObjectFile.h	Tue Aug 09 13:58:00 2011 -0700
+++ b/lib/Target/X86/X86TargetObjectFile.h	Tue Aug 09 14:31:29 2011 -0700
@@ -33,6 +33,13 @@
                             MachineModuleInfo *MMI) const;
   };
 
+// @LOCALMOD-START
+class X86_NaClElfTargetObjectFile : public TargetLoweringObjectFileELF {
+public:
+  virtual void Initialize(MCContext &Ctx, const TargetMachine &TM);
+}
+// @LOCALMOD-END
+
 } // end namespace llvm
 
 #endif
